tabular/tests/unittests/resource_allocation/test_total_resource_allocation.py

```diff
@@ -0,0 +1,228 @@
+import pytest
+import pandas as pd
+
+from autogluon.core.utils import get_cpu_count, get_gpu_count_all
+from autogluon.core.models.ensemble.bagged_ensemble_model import BaggedEnsembleModel
+from autogluon.tabular.models import AbstractModel
+
+
+class DummyBaseModel(AbstractModel):
+    
+    def __init__(self, minimum_resources={}, **kwargs):
+        self._minimum_resources = minimum_resources
+        super().__init__(**kwargs)
+        
+    def get_minimum_resources(self, **kwargs):
+        return self._minimum_resources
+    
+    def _get_default_resources(self):
+        num_cpus = 1
+        num_gpus = 1
+        return num_cpus, num_gpus
+
+
+class DummyModel(DummyBaseModel):
+    pass
+
+    
+class DummyBaggedModel(BaggedEnsembleModel):
+    pass
+
+
+def test_bagged_model_with_total_resources():
+    model_base = DummyModel()
+    bagged_model = DummyBaggedModel(model_base)
+    total_resources = {
+        'num_cpus': 1,
+        'num_gpus': 0,
+    }
+    resources = bagged_model._preprocess_fit_resources(total_resources=total_resources)
+    assert resources == total_resources
+    
+    # Given total resources more than what the system has
+    total_resources = {
+        'num_cpus': 99999,
+        'num_gpus': 99999,
+    }
+    resources = bagged_model._preprocess_fit_resources(total_resources=total_resources)
+    assert resources == {'num_cpus': get_cpu_count(), 'num_gpus': get_gpu_count_all()}
+    
+
+def test_bagged_model_with_total_resources_and_ensemble_resources():
```

Comment:
I'm assuming these tests fail if a GPU isn't present.

Have you considered overwriting `get_gpu_count_all()` for this unit test so it has no ties to the actual compute of the instance running the test? Not sure exactly the right way to do this, but Alex Shirkov would probably know.

Ditto for `get_cpu_count()`